### ISSUE TITLE
[feature] ability to always pull latest tag from registry 

### ISSUE THREAD

**Issue by a86c6f7964** [hashicorp/nomad#1197]
While I was investigating #1179 I noticed that nomad inspects the local machine, and will only pull an image from the repository if that tag is not "latest" and does not exist locally already. I use tags for things like qa and canary instead of using the "latest" tag for everything. This makes the tags more like a mutable pointer to a specific image instead of being immutable. I understand that this may be unique to my organization, so I was thinking that maybe there could be a way to configure the docker agent to treat all tags as mutable, instead of just "latest"

Something like https://github.com/novilabs/nomad/commit/66c5a8b81c5622c54404aa36ba0b37fa2f2bb1f1

**Comment by a86c6f7964** [comment]
I was thinking about this a little as well. And I like to use stuff like redis tag https://hub.docker.com/_/redis/ 3.2 This way it will get the latest patch version. If restarts do not always pull the latest tag from the registry then it would be required that all tasks specify the complete tag such as 3.2.0 I could see some people wanting to be more explicit and some not. I think that either it should be a configuration param inside the task, or inside the nomad client. I could see maybe putting it inside the task instead of the nomad client like my above commit shows.

**Comment by a86c6f7964** [comment]
Any more thought on this?

**Comment by dadgar** [comment]
@a86c6f7964 So there is a difference between evaluating the tags by the scheduler and by the client starting the task. I am okay with right before starting the task, to check the tag. However we won't have the scheduler ever reconcile the tags and update running task.

What I mean by this is if you deploy QA which maps to version 1 of your task and then QA gets updated to point at v2, all the v1 tasks would remain, Nomad wouldn't restart them just because what the label pointed to changed.

**Comment by a86c6f7964** [comment]
Yes correct.

**Comment by SkyRocknRoll** [comment]
@dadgar Right before starting the job check for any updates on the specific tag. It may be too much to ask to release this request in 0.4.0 :)

**Comment by a86c6f7964** [comment]
My branch that linked does this by configuration in the agent. But I have been thinking it probably could also make sense to be configured in the task itself. So just along side the other docker configuration would just be always pull configuration

**Comment by dadgar** [comment]
Closing in favor of older duplicate #639
